src/lib.rs

```diff
@@ -28,17 +35,26 @@ impl LogWatcher {
             Err(err) => return Err(err)
         };
 
-        let mut reader = BufReader::new(f);        
-        let pos = metadata.len();
+        let mut reader = BufReader::new(f);
+        let inode = metadata.ino();
+        let pos = if start_pos == -1 {
+            metadata.len()
+        } else {
+            if file_inode > -1 && file_inode == (inode as i64) {
+                start_pos as u64
+            } else {
+                0
```

Comment:
if we need to validate start_pos only if inode matches? Then we need to return error here. Assuming pos = 0 may cause problems when consumed.